### PR TITLE
allow for 2.x typescript compiler

### PR DESCRIPTION
**Pull request by fivetanley**

### CODE REVIEW

**Comment by fivetanley** [comment]
This also allows for the "next" tag which includes the latest dev release.

**Comment by fivetanley** [comment]
"Next" is the most forward tag available on npm afaict:

image showing tags of typescript on NPM

You can view all the `dist-tags` here: http://registry.npmjs.org/typescript

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "walk-sync": "^0.2.6"
   },
   "peerDependencies": {
-    "typescript": "^1.6.2 || ^1.7.0 || ^1.8.0 || ^1.9.0-dev"
+    "typescript": "^1.6.2 || ^1.7.0 || ^1.8.0 || ^1.9.0-dev || ^2.0.0 || next"
```
Comment (rwjblue) [inline package.json]: We should allow ^2.0.0-dev and ^2.1.0-dev also

Comment (fivetanley) [inline package.json]: good idea, done

**Pull request merged**